### PR TITLE
Master integration tests

### PR DESCRIPTION
**Pull request by ivannaranjo**
This PR is a major refactor of how the images are built to allow better testing integration moving forward.

The major feature added is to implement integration tests, by implementing two apps that conform to the integration test protocol defined at [integration testing](https://github.com/GoogleCloudPlatform/runtimes-common/blob/master/integration_tests/README.md). These two apps are:
* `test-1.0` which is designed to test the `aspnetcore-1.0` runtimes.
* `test-1.1` which is designed to test the `aspnetcore-1.1` runtimes.

The apps are stored in two different directories:
* `tests/apps/test-1.*`, these directories contain the source for the apps, the script `tests/build_apps.sh` can be used to build and publish these apps to their final destintation.
* `tests/integration_tests/test-1.*`, these directories contain the published apps, including the necessary metadata files, ready to be used for testing.

The `tools/test.sh` script is included for convenience when running the tests locally. It will deploy and run the integration tests for you in a similar way as to how it is done with the [go integration tests](https://github.com/GoogleCloudPlatform/golang-docker/tree/master/runtime-builder/tests/integration). 

The runtimes are now grouped by major version, all of them built in parallel with a single `cloudbuild.yaml` file that also includes all of the structural tests, and in the future more functional tests.

The runtime builder directory is now also updated and also built with a single `cloudbuild.yaml` that runs the structural tests and functional tests during the build process.

### CODE REVIEW

File: tools/test.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+else
+    readonly project_id=$2
+fi
+
+# Where to store the modified app definitions.
+readonly temp_builders_root=$(mktemp -d -t test_run)
+
+# Generate the builders root.
+cp $1/runtimes.yaml ${temp_builders_root}
+export readonly STAGING_BUILDER_IMAGE=gcr.io/aspnetcore-staging/aspnetcorebuild:latest
```
Comment (nkubala) [inline tools/test.sh]: do you always want to test this on `latest`?

File: builder/functional_tests/clean-1.0/clean-1.0.dll [hunk 2]
```diff
@@ -0,0 +1 @@
+Dummy file.
```
Comment (ivannaranjo) [inline builder/functional_tests/clean-1.0/clean-1.0.dll]: The runtime builder checks that the entrypoint .dll exists to ensure that the app is in a good shape before generating the Dockerfile. It does not check that the file is in fact a .dll only that it exists.

File: tools/test.sh (hunk 1 above)
Comment (ivannaranjo) [inline tools/test.sh]: Usually I would say yes, but you have a good point, I added `TAG` as away to override the tag for the builder.

**Comment by ivannaranjo** [comment]
PTAL

**Pull request merged**